skimage/morphology/_flood_fill.py

```diff
@@ -0,0 +1,274 @@
+"""flood_fill.py - inplace flood fill algorithm
+
+This module provides a function to fill all equal (or within tolerance) values
+connected to a given seed point with a different value.
+"""
+
+import numpy as np
+import warnings
+
+from .extrema import (_resolve_neighborhood, _set_edge_values_inplace,
+                      _fast_pad)
+from .watershed import _offsets_to_raveled_neighbors
+from ._flood_fill_cy import _flood_fill_equal, _flood_fill_tolerance
+
+
+def flood_fill(image, seed_point, new_value, *, selem=None, connectivity=None,
+               tolerance=None, inplace=False):
+    """Perform flood filling on an image.
+
+    Starting at a specific `seed_point`, connected points equal or within
+    `tolerance` of the seed value are found, then set to `new_value`.
+
+    Parameters
+    ----------
+    image : ndarray
+        An n-dimensional array.
+    seed_point : tuple or int
+        The point in `image` used as the starting point for the flood fill.  If
+        the image is 1D, this point may be given as an integer.
+    new_value : `image` type
+        New value to set the entire fill.  This must be chosen in agreement
+        with the dtype of `image`.
+    selem : ndarray, optional
+        A structuring element used to determine the neighborhood of each
+        evaluated pixel. It must contain only 1's and 0's, have the same number
+        of dimensions as `image`. If not given, all adjacent pixels are
+        considered as part of the neighborhood (fully connected).
+    connectivity : int, optional
+        A number used to determine the neighborhood of each evaluated pixel.
+        Adjacent pixels whose squared distance from the center is larger or
+        equal to `connectivity` are considered neighbors. Ignored if
+        `selem` is not None.
+    tolerance : float or int, optional
+        If None (default), adjacent values must be strictly equal to the
```

Comment:
This is not quite accurate: By default, in order to be filled, adjacent values must be strictly equal to the value of `image` at `seed_point`.